From: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Date: Mon, 26 Jun 2023 13:55:27 +0300
Subject: ACPI: scan: Move acpi_root to internal header
Patch-mainline: v6.5-rc1
Git-commit: 7ba6b73db3dbe6cf365a8122e4ce36559a7714cc
References: jsc#PED-6741

Compiler is not happy about handling of acpi_root variable:

  ...drivers/acpi/bus.c:37:20: warning: symbol 'acpi_root' was not declared. Should it be static?

Move it's definition to the internal header.

Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 drivers/acpi/internal.h |    2 ++
 drivers/acpi/scan.c     |    2 --
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/acpi/internal.h
+++ b/drivers/acpi/internal.h
@@ -11,6 +11,8 @@
 
 #include <linux/idr.h>
 
+extern struct acpi_device *acpi_root;
+
 int early_acpi_osi_init(void);
 int acpi_osi_init(void);
 acpi_status acpi_os_initialize1(void);
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -24,8 +24,6 @@
 
 #include "internal.h"
 
-extern struct acpi_device *acpi_root;
-
 #define ACPI_BUS_CLASS			"system_bus"
 #define ACPI_BUS_HID			"LNXSYBUS"
 #define ACPI_BUS_DEVICE_NAME		"System Bus"
